libmscore/cursor.h

```diff
@@ -96,18 +97,34 @@ class Cursor : public QObject {
 
       int qmlKeySignature();
 
-      //@ rewind cursor
-      //@   type=0      rewind to start of score
-      //@   type=1      rewind to start of selection
-      //@   type=2      rewind to end of selection
-      Q_INVOKABLE void rewind(int type);
-
+      //@ moves cursor to first element of current track in score (filtered by 'filter')
+      Q_INVOKABLE void scoreStart();
+      //@ moves cursor to last element of current track in score (filtered by 'filter')
+      Q_INVOKABLE void scoreEnd();
+      //@ moves cursor to first element of current track in selection (filtered by 'filter');
+      //@ if there is no selection, same as scoreStart()
+      Q_INVOKABLE void selectionStart();
+      //@ moves cursor to last element of current track in selection (filtered by 'filter');
+      //@ if there is no selection, same as scoreEnd()
+      Q_INVOKABLE void selectionEnd();
+      //@ moves cursor to next element of current track (filtered by 'filter')
       Q_INVOKABLE bool next();
+      //@ moves cursor to first segment of current track in next measure (filtered by 'filter');
+      //@ returns false if end of score is reached
       Q_INVOKABLE bool nextMeasure();
-      Q_INVOKABLE void add(Ms::Element*);
 
+      //@ adds an Element to the current cursor segment in the current track
+      Q_INVOKABLE void add(Ms::Element*);
+      //@ adds a note of pitch 'pitch' to the chord in current track of current segment;
+      //@ chord must already exist
       Q_INVOKABLE void addNote(int pitch);
 
+      // rewind cursor
+      //   type=0      rewind to start of score
+      //   type=1      rewind to start of selection
+      //   type=2      rewind to end of selection
+      Q_INVOKABLE void rewind(int type);  // obsolete; replaced by scoreStart/End() and selectionStart/End():
+
```

Comment:
I probably wouldn't remove the description completely, because rewind() can still be found in existing plugins.
Besides: If you don't intent to use the Cursor for iterating, but only to find the selection, using selectionStart() and selectionEnd() to retrieve the first track, last track, first segment and last segment of the selection doesn't seem to be a lot easier than using rewind().